mountpoint-s3/src/prefetch.rs

```diff
@@ -161,15 +162,7 @@ where
             "read"
         );
 
-        // Currently, we set preferred part size to the current read size.
-        // Our assumption is that the read size will be the same for most sequential
-        // read and it can be aligned to the size of prefetched chunks.
-        //
-        // We initialize this value to 128k as it is the Linux's readahead size
-        // and it can also be used as a lower bound in case the read size is too small.
-        // The upper bound is 1MiB since it should be a common IO size.
-        let max_preferred_part_size = 1024 * 1024;
-        self.preferred_part_size = self.preferred_part_size.max(length).min(max_preferred_part_size);
+        self.max_read_request_size = self.max_read_request_size.max(length);
```

Comment:
It's a bit weird to me that the clamping happens in the part feed rather than here, which is where we already know about readahead and the like. Is there some reason the part feed shouldn't just take and trust whatever part size this code asks for (with the new catch that it might choose to ignore it)?

Reply:
Fair point about the readahead. In theory, clamping early looses information but I don't actually see how it matters in practice. I'll rework.